### PR TITLE
legacy adapter config migration

### PR DESCRIPTION
**Pull request by Florian-Limpoeck**

### CODE REVIEW

File: hivemq-edge/src/main/java/com/hivemq/protocols/ProtocolAdapterManager.java [hunk 1]
```diff
@@ -171,10 +172,48 @@ public ProtocolAdapterManager(
             }
         }
 
+        final Map<String, Object> allAdapters = new HashMap<>();
+        for (ProtocolAdapterWrapper<? extends ProtocolAdapter> value : protocolAdapters.values()) {
```
Comment (YannickWeber) [inline hivemq-edge/src/main/java/com/hivemq/protocols/ProtocolAdapterManager.java:176]:
```suggestion
        for (final ProtocolAdapterWrapper<? extends ProtocolAdapter> value : protocolAdapters.values()) {
```

File: modules/hivemq-edge-module-plc4x/src/main/java/com/hivemq/edge/adapters/plc4x/types/ads/ADSProtocolAdapterFactory.java [hunk 2]
```diff
@@ -42,4 +51,47 @@ public class ADSProtocolAdapterFactory implements ProtocolAdapterFactory<ADSAdap
         return ADSAdapterConfig.class;
     }
 
+    @Override
+    public @NotNull ADSAdapterConfig convertConfigObject(
+            final @NotNull ObjectMapper objectMapper, final @NotNull Map<String, Object> config) {
+        if(config.get("adsToMqtt") != null || config.get("mqttToAds") != null) {
```
Comment (YannickWeber) [inline modules/hivemq-edge-module-plc4x/src/main/java/com/hivemq/edge/adapters/plc4x/types/ads/ADSProtocolAdapterFactory.java:57]: formatting

File: modules/hivemq-edge-module-plc4x/src/main/java/com/hivemq/edge/adapters/plc4x/types/siemens/S7ProtocolAdapterFactory.java [hunk 3]
```diff
@@ -43,4 +51,49 @@ public class S7ProtocolAdapterFactory implements ProtocolAdapterFactory<S7Adapte
         return S7AdapterConfig.class;
     }
 
+
+    @Override
+    public @NotNull S7AdapterConfig convertConfigObject(
+            final @NotNull ObjectMapper objectMapper, final @NotNull Map<String, Object> config) {
+        if(config.get("s7ToMqtt") != null || config.get("mqttToS7") != null) {
```
Comment (YannickWeber) [inline modules/hivemq-edge-module-plc4x/src/main/java/com/hivemq/edge/adapters/plc4x/types/siemens/S7ProtocolAdapterFactory.java:58]: formatting

**Review by YannickWeber — APPROVED** [review]

File: modules/hivemq-edge-module-file/src/main/java/com/hivemq/edge/adapters/file/FileProtocolAdapterFactory.java [hunk 4]
```diff
@@ -30,14 +38,51 @@ public class FileProtocolAdapterFactory implements ProtocolAdapterFactory<FileAd
...
+    public @NotNull ProtocolAdapter createAdapter(
+            final @NotNull ProtocolAdapterInformation adapterInformation,
+            @NotNull final ProtocolAdapterInput<FileAdapterConfig> input) {
         return new FilePollingProtocolAdapter(adapterInformation, input);
     }
 
+    @Override
+    public @NotNull FileAdapterConfig convertConfigObject(
+            final @NotNull ObjectMapper objectMapper, final @NotNull Map<String, Object> config) {
+        if (config.get("FileToMqtt") != null || config.get("mqttToFile") != null) {
```
Comment (Florian-Limpoeck) [inline modules/hivemq-edge-module-file/src/main/java/com/hivemq/edge/adapters/file/FileProtocolAdapterFactory.java:50]: this wont work ;) must be:
```suggestion
        if (config.get("fileToMqtt") != null || config.get("mqttToFile") != null) {
```

File: modules/hivemq-edge-module-http/src/main/java/com/hivemq/edge/adapters/http/HttpProtocolAdapterFactory.java [hunk 5]
```diff
@@ -39,9 +46,47 @@ public class HttpProtocolAdapterFactory implements ProtocolAdapterFactory<HttpAd
         return new HttpProtocolAdapter(adapterInformation, input);
     }
 
+    @Override
+    public @NotNull HttpAdapterConfig convertConfigObject(
+            final @NotNull ObjectMapper objectMapper, final @NotNull Map<String, Object> config) {
+        if (config.get("HttpToMqtt") != null || config.get("mqttToHttp") != null) {
```
Comment (Florian-Limpoeck) [inline modules/hivemq-edge-module-http/src/main/java/com/hivemq/edge/adapters/http/HttpProtocolAdapterFactory.java:52]:
```suggestion
        if (config.get("httpToMqtt") != null || config.get("mqttToHttp") != null) {
```

**Pull request merged**